hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemMkDir.java

```diff
@@ -167,4 +181,107 @@ public void testMkdirWithExistingFilename() throws Exception {
     intercept(FileAlreadyExistsException.class, () -> fs.mkdirs(new Path("/testFilePath")));
     intercept(FileAlreadyExistsException.class, () -> fs.mkdirs(new Path("/testFilePath/newDir")));
   }
+
+  /**
+   * Test mkdirs with HDI folder configuration,
+   * verifying the correct header and directory state.
+   */
+  @Test
```

Comment:
Add test for other variations such as ALL Caps, only one later caps, all small. Also one test can be when we set multiple metadata keys, the directory one is correctly identified.